### ISSUE TITLE
Minimizing group leaves gap in the top

### ISSUE THREAD

**Issue by Lazaroth** [olifolkerd/tabulator#672]
I'm using groups. If I scroll down a while and minimize a group, then scroll all the way back up, the height that was minimized is shown as empty space in the very top.

**Comment by mikeshow** [comment]
Hi,
I noticed that too. And if I scroll up with the mouse all the way to the top, the gap disappears.

**Comment by olifolkerd** [comment]
could you post a copy of your table constructor please, also which browser are you seeing this on?

**Comment by mikeshow** [comment]
Hi Oli,

The browsers:
chrome 62, chrome canary 64, firefox 56.0.1(64bit), opera 48.0, edge 40.xxxx. I didn't bother to check with IE.

So in my case:

I do the call to fill tabulator. At first, all the groups are closed.
1st case: I scroll down, open a group, rowClick a line, arrive in the update form, click 'cancel' button -> tabulator call -> scrollToRow selected, scroll up with the mouse and everything's good.
2nd case: I open the first group and then do exactly as in the 1st case. When I come back and scroll up with the mouse, then there is a blank area, apparently the size of the open group.

Tabulator:

```javascript
      var tbltr = $("#thetable");
      tbltr.tabulator({
         layout: 'fitData',
         sortable: false,
         virtualDom: true,
         height: tabulHeight,
         placeholder: i18n[current_language].t_no_data,
         groupBy: 'categ_groupby',
         groupStartOpen: false,
         groupToggleElement: "header",
         groupHeader: function (value, count, data) {
            return data[0].categ_name;
         },
         groupClosedShowCalcs: true,
         tooltipsHeader: true,
         rowFormatter: function (row) {
            var categparent = row.getData().categ_parent;
            if (categparent === "0")
            {
               row.getElement().css({"background-color": "#f4f4f4"});
            }
         },
         locale: true,
         langs: {
            "en-us": {
               "ajax": {
                  "loading": i18n[current_language].tbltr_loading,
                  "error": i18n[current_language].tbltr_error
               },
               "headerFilters": {
                  "default": i18n[current_language].tbltr_filer_def
               }
            },
            "en-gb": {
               "ajax": {
                  "loading": i18n[current_language].tbltr_loading,
                  "error": i18n[current_language].tbltr_error
               },
               "headerFilters": {
                  "default": i18n[current_language].tbltr_filer_def
               }
            }
         },
         footerElement: $('<div class="tabulator-footer"><button id="category_new" class="mcdmcquote-form-submit">' + i18n[current_language].btn_new + '</button></div>'),
         columns: [
            {
               title: i18n[current_language].t_name_categ,
               field: "categ_name",
               align: "left",
               width: widthCol1,
               headerSort: false,
               formatter: function (cell, formatParams) {
                  var categparent = cell.getRow().getData().categ_parent;
                  if (categparent === "0")
                  {
                     return cell.getRow().getData().categ_name;
                  }
                  return '';
               },
               bottomCalc: "count",
               headerFilter: true
            },
            {
               title: i18n[current_language].t_name_item,
               field: "item_name",
               align: "left",
               width: widthCol1,
               headerSort: false,
               headerFilter: true
            },
            {
               title: '<i class="fa fa-plus-circle fa-lg" aria-hidden="true"></i>',
               align: 'center',
               width: 32,
               headerSort: false,
               headerTooltip: i18n[current_language].t_h_add,
               tooltip: function (cell) {
                  var categparent = cell.getRow().getData().categ_parent;
                  if (categparent === "0")
                  {
                     return i18n[current_language].t_clic_nw_item;
                  }
                  return '';
               },
               formatter: function (cell, formatParams) {
                  var categparent = cell.getRow().getData().categ_parent;
                  if (categparent === "0")
                  {
                     return '<i class="fa fa-plus-circle fa-lg" aria-hidden="true"></i>';
                  }
                  return '';
               },
               cellClick: function (e, cell) {
                  var categparent = cell.getRow().getData().categ_parent;
                  if (categparent === "0")
                  {
                     var categ_id = cell.getRow().getData().id;
                     rowIdToScroll = categ_id;
                     showView('tabvw_ins_item', categ_id);
                  }
                  e.stopPropagation();
               }
            },
            {
               title: '<i class="fa fa-trash-o fa-lg" aria-hidden="true"></i>',
               align: 'center',
               width: 32,
               headerSort: false,
               headerTooltip: i18n[current_language].t_h_del,
               tooltip: function (cell) {
                  var categparent = cell.getRow().getData().categ_parent;
                  if (categparent === "0")
                  {
                     return i18n[current_language].t_del_categories;
                  }
                  return i18n[current_language].t_del_items;
               },
               formatter: function (cell, formatParams) {
                  var categparent = cell.getRow().getData().categ_parent;
                  if (categparent === "0")
                  {
                     return '<i class="fa fa-trash-o fa-lg" aria-hidden="true"></i>';
                  }
                  return '<i class="fa fa-eraser" aria-hidden="true"></i>';
               },
               cellClick: function (e, cell) {
                  var catitem_id = cell.getRow().getData().id;
                  var categparent = cell.getRow().getData().categ_parent;
                  var categname = cell.getRow().getData().categ_name;
                  var itemname = cell.getRow().getData().item_name;
                  rowIdToScroll = catitem_id;
                  if (categparent === "0")
                  {
                     delete_selected_category(catitem_id, categname, e);
                  } else
                  {
                     delete_selected_item(catitem_id, itemname, e);
                  }
                  e.stopPropagation();
               }
            }
         ],
         rowClick: function (e, row) {
            // request the infos to update.
            var id = row.getIndex();
            var categparent = row.getData().categ_parent;
            // check if not clicked the bottom calc
            if (id !== undefined)
            {
               rowIdToScroll = id;
               $.ajax({
                  type: 'POST',
                  url: mcdmcdash.ajaxurl,
                  data: {
                     action: 'view_request_category_server_side',
                     categ_nonce: mcdmcdash.categ_nonce,
                     do_what: 'request',
                     dmc_id: id
                  },
                  success: function (response) {
                     if (response.success)
                     {
                        if (categparent === "0")
                        {
                           showView('tabvw_updt_categ', id, response.data.content);
                        } else
                        {
                           showView('tabvw_updt_item', id, response.data.content);
                        }
                        return false;
                     } else
                     {
                        var themessage = i18n[current_language].t_no_data;
                        if (typeof response.data.message === 'string')
                        {
                           themessage = response.data.message;
                        }
                        $.notify(themessage, {position: 'top right', className: 'warn'});
                        console.log('error: ' + response.data.message);
                     }
                  },
                  error: function (error_msg) {
                     $.notify(error_msg, {position: 'top right', className: 'error'});
                     console.log('error: ' + error_msg);
                  }
               });
            }
            return false;
         },
         ajaxResponse: function (url, params, response) {
            return response.data.content;
         },
         renderComplete: function () {
            if (rowIdToScroll > 1)
            {
               tbltr.tabulator("scrollToRow", rowIdToScroll);
            }
         }
      });
```


The call:

```javascript
   var tbltr = $("#thetable");
   tbltr.tabulator(
      "setData",
      mcdmcdash.ajaxurl,
      {
         action: 'tabulator_categories_server_side',
         categ_nonce: mcdmcdash.categ_nonce,
         do_what: 'list'
      },
      "POST"
   );
```

Cheers,

Mike.

**Comment by olifolkerd** [comment]
i will ensure the rendering engine is improved to handle this scenario in the next release.

cheers

Oli

**Comment by nperree** [comment]
Hi Oli,

I have another scenario which produces this which I think might help:

    <script>
        $(document).ready(function() {

            function rowClick(e, row) {

                var grid = $('#grid'),
                    grid2 = $('#grid2');

                if (grid2.hasClass('hidden')) {
                    // Show the second grid

                    grid.tabulator('setHeight', 'calc(100% - 30%)');
                    grid2.tabulator('setHeight', '30%');
                    grid2.removeClass('hidden');
                } else {
                    // Hide the second grid

                    grid.tabulator('setHeight', '100%');
                    grid2.addClass('hidden');
                }

                grid.tabulator('redraw', true);
                grid2.tabulator('redraw', true);

                // Removing this line prevents the issue
                row.scrollTo();
            }

            var columns, data, config, i;

            // Create a number of columns
            columns = [{
                    title: 'Column 1',
                    field: 'data1',
                    width: '20%'
                },
                {
                    title: 'Column 2',
                    field: 'data2',
                    width: '20%'
                },
                {
                    title: 'Column 3',
                    field: 'data3',
                    width: '20%'
                },
                {
                    title: 'Column 4',
                    field: 'data4',
                    width: '20%'
                },
                {
                    title: 'Column 5',
                    field: 'data5',
                    width: '20%'
                },
                {
                    title: 'Column 6',
                    field: 'data6',
                    width: '20%'
                },
                {
                    title: 'Column 7',
                    field: 'data7',
                    width: '20%'
                },
                {
                    title: 'Column 8',
                    field: 'data8',
                    width: '20%'
                }
            ];

            data = [];

            // Add fake data
            for (i = 0; i < 100; i = i + 1) {
                data.push({
                    data1: 1,
                    data2: 2,
                    data3: 3,
                    data4: 4,
                    data5: 5,
                    data6: 6,
                    data7: 7,
                    data8: 8
                });
            }

            // Create the main table configuration
            config = {
                layout: 'fitColumns',
                height: '100%',
                rowClick: rowClick,
                columns: columns
            };

            // Setup the second grid
            $('#grid').tabulator(config);
            $('#grid').tabulator('setData', data);

            $('#grid2').tabulator({
                layout: 'fitColumns',
                columns: [{
                        title: 'Col 1',
                        width: '50%'
                    },
                    {
                        title: 'Col 2',
                        width: '50%'
                    }
                ]
            });

        });

    </script>

When you scroll to the bottom of the top grid and click on a row, the second grid is displayed.  If you scroll to the top, the gap can be seen.  If you then scroll down again and back up, there is a slight delay and the top grid gets drawn properly again.

The rowClick() event handler includes a scrollTo() call on the row which was clicked.  If this line is removed, the top grid is redrawn without any scroll applied but the gap does not appear.

I have been using Tabulator 3.3.2 and chrome for testing.

I hope this helps with this.

**Comment by olifolkerd** [comment]
Hey All,

I have just pushed a fix to the 3.4 branch to resolve the whitespace issue, It will be released later today.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Tabulator version 3.4 has now been officially released!

You can find details of the new release in the [News Section](http://tabulator.info/news)

Full documentation of all the new features can be found [Here](http://tabulator.info/docs/3.4)

Cheers

Oli :)
